src/model/link.rs

```diff
@@ -79,7 +191,9 @@ impl<S: Read + Write> Stream<S> {
     /// Only works when stream is a SSL stream
     pub fn shutdown(&mut self) -> RdpResult<()> {
         Ok(match self {
-            Stream::Ssl(e) => e.shutdown()?,
+            Stream::Ssl(stream) => {
+                stream.conn.send_close_notify();
+            }
             _ => (),
         })
```

Comment:
To make sure that this is sent and writer is flushed at the end I've added this call `stream.flush()?`. 
It'll send any remaining tls messages, including the one we just queued, via this method: 

https://github.com/rustls/rustls/blob/8e5968f0a25b6319488d22dcd67bab62dfb4f713/rustls/src/stream.rs#L137

and this method is completing all io and all tls wrties via this call:

https://github.com/rustls/rustls/blob/8e5968f0a25b6319488d22dcd67bab62dfb4f713/rustls/src/conn.rs#L462

I think after that flush we can be sure it will be sent if there is no errors.

Reply:
👍